corehq/apps/app_manager/static/app_manager/js/releases.js

```diff
@@ -161,6 +161,7 @@ hqDefine('app_manager/js/releases.js', function () {
             ga_track_event('App Manager', 'Deploy Button', self.id());
             analytics.workflow('Clicked Deploy');
             $.post(releasesMain.options.urls.hubspot_click_deploy);
+            self.get_short_odk_url();
```

Comment:
This resolves the workflow where the user clicks "Deploy", then clicks "Send to phone via SMS", then clicks "Close". 

`get_short_code_url()` calls `generate_short_url()` (if necessary) with the right `url_type` (i.e. Java / Android / Android + media) which in turn sets the Observable `app_code()`.

Toggling "Include media" will also set `app_code()` correctly.

It's `app_code()` that gets rendered.

But the "Deploy" button (in the case of this line of code) and the "Download to Android" accordion panel don't reset `app_code()` if it is storing a URL for Java. This PR should resolve that.

I should have written all that up in the PR comment. I hope this clarifies it.